AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/TestSupport/SyncEngineTestBase.swift

```diff
@@ -47,41 +48,47 @@ class SyncEngineTestBase: XCTestCase {
         }
     }
 
-    /// Sets up a StorageAdapter backed by an in-memory SQLite database
-    func setUpStorageAdapter() throws {
+    /// Sets up a StorageAdapter backed by an in-memory SQLite database. Optionally registers and sets up models in
+    /// `models`.
+    func setUpStorageAdapter(preCreating models: [Model.Type] = []) throws {
+        models.forEach { ModelRegistry.register(modelType: $0) }
```

Comment:
This is a good point. My original intent here was to allow specificity if such is required, and sensible defaults otherwise. But that's really a bandaid over the real problem: Because we were moving fast, I didn't do a good job of separating "unit" from "functional" tests, which means a lot of the "unit" tests actually depend on a lot of system setup--like these models.

Longer term, my instinct is that we should  refactor the "unit" tests into actual standalone tests of each submodule, like how I kind of started to do with the InitialSyncOrchestrator tests, having no interdependencies on other system components. Then we'd put together have a suite of functional tests to exercise the interdependencies between components (separate from the integration tests, which also test the external systems). This method would be part of the functional suite, and would allow tests to specify the models for the various conditions they want to test as part of the overall system behavior.

I'm open to suggestions. For now, leaving this here provides some assurance of the behavior, at the cost of adding technical test debt every time we re-use this method or copy/paste the pattern.